JASP-Engine/JASP/R/bayesianmetaanalysis.R

```diff
@@ -0,0 +1,1760 @@
+  }
+
+  if(options$modelSpecification == "CRE" && options$direction == "allPos"){
+    xlimLeft <- 0
+  } else if(options$modelSpecification == "CRE" && options$direction == "allNeg"){
+    xlimRight <- 0
+  }
+
+  xlimRight <- mean + (s * 5)  
+  xlimLeft <- xlimLeft - 0.05
+  xlimRight <- xlimRight + 0.05
+  
+  # Create dataframe for ggplot
+  x <- c(xlimLeft, xlimRight)
+  df <- data.frame(x = x)
+  
+  xBreaks <- JASPgraphs::getPrettyAxisBreaks(seq(xlimLeft, xlimRight, 0.5))
+  
+  # Plot density function
+  plot <- ggplot2::ggplot(df, ggplot2::aes(x)) +
+          ggplot2::stat_function(fun = prior, n = 1000, size = 1) +
+          ggplot2::labs(x = xlab, y = gettext("Density")) +
+          ggplot2::xlim(xlimLeft, xlimRight) +
+          ggplot2::scale_x_continuous(breaks = xBreaks)
+  plot <- JASPgraphs::themeJasp(plot)
+  priorPlot$plotObject <- plot
+  return()
+}
+
+# Plot: Prior and Posterior
+.bmaPriorAndPosteriorPlot <- function(jaspResults, dataset, options, ready, .bmaDependencies) {
+  postContainer <- createJaspContainer(title = gettext("Prior and Posteriors"))
+  postContainer$dependOn(c(.bmaDependencies, "plotPosterior", "shade", "addInfo", "addLines"))
+  jaspResults[["postContainer"]] <- postContainer
+  jaspResults[["postContainer"]]$position <- 5
+  
+  # Create empty plot
+  postPlotES <- createJaspPlot(plot = NULL, title = gettext("Effect size"), width = 500, height = 350)
+  postPlotES$position <- 1
+  
+
+  
+  # Check if ready
+  if(!ready){
+    return()
+  }
+  
+  # Fill posterior plot effect size
+  .bmaFillPostPlot(postPlotES, jaspResults, dataset, options, type = "ES")
+  postContainer[["ES"]] <- postPlotES
+  
+  # Make posterior plot heterogeneity
+  if(options$modelSpecification != "FE"){
+    postPlotSE <- createJaspPlot(plot = NULL, title = gettext("Heterogeneity"), width = 500, height = 350)
+    postPlotSE$position <- 2
+    postContainer[["SE"]] <- postPlotSE
+    .bmaFillPostPlot(postPlotSE, jaspResults, dataset, options, type = "SE")
+  }
+}
+
+# Fill prior and posterior plot
+.bmaFillPostPlot <- function(postPlot, jaspResults, dataset, options, type){
+  # Get results from jasp state
+  bmaResults <- .bmaResultsState(jaspResults, dataset, options, .bmaDependencies)
+  
+  # Get prior and posterior functions, and 95% CI intervals
+  alpha <- 0.2
+  postName <- "Posterior"
+  valuesCol <- c("black", "black")
+  valuesLine <- c("solid", "dotted")
+  
+  if(type == "ES"){
+    xlab <- bquote(paste(.(gettext("Effect size")), ~mu))
+    xlim <- c(-4, 4)
+    if(options[["modelSpecification"]] == "BMA"){
+      int <- c(bmaResults[["bma"]]$estimates["averaged", "2.5%"], bmaResults[["bma"]]$estimates["averaged", "97.5%"])
+      postName <- "Averaged"
+      if(options[["addLines"]]){
+        labelsModel <- c(bquote(paste(.(gettext("Fixed H")), [1])), bquote(paste(.(gettext("Random H")), [1])), bquote(paste(.(gettext("Averaged H")), [1])), bquote(paste(.(gettext("Prior H")), [1])))
```

Comment:
This line is not valid R code, according to travis: `Error: unexpected '[' in "c(bquote(paste(.(gettext("Fixed H")), ["`